File: server/FuelFlow.Application/DTOs/Auth/RegisterRequest.cs
namespace FuelFlow.Application.DTOs.Auth;

/// <summary>
/// The request payload for owner self-registration.
///
/// Registration creates Owner user only.
/// Organization and first station are added during onboarding after first login.
///
/// Phone-first per [M01-F09-R01]: phone is required, email is optional.
/// When email is provided it must be unique ([M01-F01-R01]) and a verification
/// email is dispatched alongside the SMS OTP.
/// </summary>
public class RegisterRequest
{
    // Owner info
    public string FullName { get; set; } = string.Empty;

    /// <summary>Optional email. When provided, must be unique and will receive a verification link.</summary>
    public string? Email { get; set; }

    /// <summary>Required Pakistani phone in <c>+92XXXXXXXXXX</c> format.</summary>
    public string Phone { get; set; } = string.Empty;

    public string Password { get; set; } = string.Empty;

    /// <summary>
    /// Optional device identifier for session tracking (e.g. browser fingerprint).
    /// </summary>
    public string? DeviceId { get; set; }
}

File: fuel-flow-web/src/lib/api/auth/registerRequest.ts
/**
 * Auth API — register, login, etc.
 */

import { api } from '../client'

export interface RegisterRequest {
  fullName: string
  /** Optional — when provided, must be a valid email; backend enforces uniqueness. */
  email?: string
  phone: string
  password: string
  deviceId?: string
}

export interface RegisterResponse {
  success: boolean
  message: string
}

/** API response wrapper for register endpoint */
export interface RegisterApiResponse {
  success: boolean
  data: RegisterResponse
}

/**
 * Register a new owner account.
 * Organization and first station are added during onboarding after first login.
 */
export async function register(
  payload: RegisterRequest
): Promise<RegisterApiResponse> {
  return api.post<RegisterApiResponse>('/auth/register', payload)
}
